src/drm.cpp

```diff
@@ -2261,23 +2258,27 @@ bool drm_set_mode( struct drm_t *drm, const drmModeModeInfo *mode )
 		return false;
 
 	drm_log.infof("selecting mode %dx%d@%uHz", mode->hdisplay, mode->vdisplay, mode->vrefresh);
-
 	drm->pending.mode_id = mode_id;
 	drm->needs_modeset = true;
 
-	g_nOutputWidth = mode->hdisplay;
-	g_nOutputHeight = mode->vdisplay;
 	g_nOutputRefresh = mode->vrefresh;
 
 	// Auto-detect portrait mode
 	g_bRotated = g_nOutputWidth < g_nOutputHeight;
 
-	if ( g_bRotated )
+	if ( g_drmEffectiveOrientation == PANEL_ORIENTATION_90 || g_drmEffectiveOrientation == PANEL_ORIENTATION_270 )
+    {
+        g_nOutputWidth = mode->vdisplay;
+        g_nOutputHeight = mode->hdisplay;
+    }
+	else
 	{
-		g_nOutputWidth = mode->vdisplay;
-		g_nOutputHeight = mode->hdisplay;
+		g_nOutputWidth = mode->hdisplay;
+		g_nOutputHeight = mode->vdisplay;
 	}
 
+	update_drm_effective_orientation(drm, drm->connector);
```

Comment:
We need to call `update_drm_effective_orientation()` before checking `g_drmEffectiveOrientation` in the `if` above I believe…

Reply:
That's what I thought, I ended up with unexpected behavior doing that though. On my laptop Steam rendered on the bottom left hand side of the screen. I called it after and it seemed to be fine. While tinkering with the logic and running it after it seemed to be updating the display so I was thinking for some reason this was how it would need to be updated.

Reply:
Laptop display with `update_effective_orientation' ran **before** the if statement.
![PXL_20221012_112722389.jpg](https://user-images.githubusercontent.com/16360335/195331098-65d5e626-69f6-4851-bf0b-7401e2db6cd3.jpg)


Rotated display OneXPlayer with statement ran **after**

![PXL_20221012_023311949.jpg](https://user-images.githubusercontent.com/16360335/195331180-756dcf02-c6bd-490e-9a94-f25b373ac26c.jpg)

It works fine when I don't try and control the `vdisplay` or ``hdisplay` values for both systems..

Reply:
`g_bRotated` above reads previous values of `g_nOutputWidth`/`g_nOutputHeight`.